Makefile

```diff
@@ -3,12 +3,26 @@ COMPARE ?= 0
 
 ifeq ($(CC),)
 HOSTCC := gcc
+# use system clang on macOS; GCC is wonky w/ homebrew sometimes
+ifneq ($(OS),Windows_NT)
+UNAME := $(shell uname -s)
+ifeq ($(UNAME),Darwin)
+HOSTCC := /usr/bin/clang
```

Comment:
No. Just no.

Reply:
uh, yes. the `gcc` command is ambiguous asf on macOS. we should stick to the system compiler unless otherwise specified